src/core/pointcloud/qgspointclouddataprovider.cpp

```diff
@@ -53,6 +55,32 @@ QgsGeometry QgsPointCloudDataProvider::polygonBounds() const
   return QgsGeometry::fromRect( extent() );
 }
 
+void QgsPointCloudDataProvider::generateStatistics()
+{
+  QgsPointCloudIndex *index = this->index();
+  mStatisticsGenerationState = PointCloudStatisticsGenerationState::Calculating;
+  mStatsCalculator.reset( new QgsPointCloudStatsCalculator( index ) );
+  QVector<QgsPointCloudAttribute> attributes = index->attributes().attributes();
+  for ( int i = 0; i < attributes.size(); ++i )
+  {
+    QString name = attributes.at( i ).name();
+    if ( name == QStringLiteral( "X" ) || name == QStringLiteral( "Y" ) || name == QStringLiteral( "Z" ) )
```

Comment:
it's worth commenting that we skip X/Y/Z because min/max of those are always available in metadata... (as this may change if we need more advanced stats, e.g. histogram)